### PR TITLE
Keep database transactions not too big

### LINKED ISSUE #13255: Issue with recent Migration[156]
- Gitea version (or commit ref): [`Gitea version 1.14.0+dev-49-g327f18c3b built with GNU Make 4.3, go1.15.3 : bindata, sqlite, sqlite_unlock_notify`](327f18c3b)
- Git version: `git version 2.29.0.311.g38ddec7408`
- Operating system: Arch Linux
- Built from source
  - [x] Yes
- Run with
  - [x] systemd
- Database:
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] No
- Log gist:
```log
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 cmd/web.go:110:runWeb() [I] Starting Gitea on PID: 2637956
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...s/setting/setting.go:832:NewContext() [I] Default UI Location is Europe/Vienna
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...dules/setting/git.go:91:newGit() [I] Git Version: 2.29.0.311.g38ddec7408, Wire Protocol Version 2 Enabled
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 cmd/web.go:151:runWeb() [I] Global init
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...s/setting/setting.go:832:NewContext() [I] Default UI Location is Europe/Vienna
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...dules/setting/git.go:91:newGit() [I] Git Version: 2.29.0.311.g38ddec7408, Wire Protocol Version 2 Enabled
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 routers/init.go:164:GlobalInit() [T] AppPath: /usr/bin/gitea
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 routers/init.go:165:GlobalInit() [T] AppWorkPath: /var/lib/gitea
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 routers/init.go:166:GlobalInit() [T] Custom path: /var/lib/gitea/custom
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 routers/init.go:167:GlobalInit() [T] Log path: /var/log/gitea/
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...dules/setting/log.go:297:newLogService() [I] Gitea v1.14.0+dev-53-gdafc634a4 built with GNU Make 4.3, go1.15.3 : bindata, sqlite, sqlite_unlock_notify
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...dules/setting/log.go:343:newLogService() [I] Gitea Log Mode: Console(Console:info)
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...dules/setting/log.go:249:generateNamedLogger() [I] Macaron Log: Console(console:info)
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...dules/setting/log.go:249:generateNamedLogger() [I] Router Log: Console(console:info)
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...dules/setting/log.go:249:generateNamedLogger() [I] Xorm Log: Console(console:info)
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...les/setting/cache.go:70:newCacheService() [I] Cache Service Enabled
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...les/setting/cache.go:81:newCacheService() [I] Last Commit Cache Service Enabled
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...s/setting/session.go:63:newSessionService() [I] Session Service Enabled
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...es/setting/mailer.go:109:newMailService() [I] Mail Service Enabled
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...es/setting/mailer.go:131:newNotifyMailService() [I] Notify Mail Service Enabled
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...s/storage/storage.go:151:initAttachments() [I] Initialising Attachment storage with type:
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...les/storage/local.go:44:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/attachments
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...s/storage/storage.go:145:initAvatars() [I] Initialising Avatar storage with type:
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...les/storage/local.go:44:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/avatars
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...s/storage/storage.go:163:initRepoAvatars() [I] Initialising Repository Avatar storage with type:
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...les/storage/local.go:44:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/repo-avatars
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...s/storage/storage.go:157:initLFS() [I] Initialising LFS storage with type:
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...les/storage/local.go:44:NewLocalStorage() [I] Creating new Local Storage at /var/lib/gitea/data/lfs
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 routers/init.go:72:initDBEngine() [I] Beginning ORM engine initialization.
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #1/15...
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:29:36 nebula gitea[2637956]: 2020/10/22 08:29:36 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:29:48 nebula gitea[2637956]: 2020/10/22 08:29:48 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #1/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:29:48 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:29:48 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:29:48 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:29:48 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:29:48 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:29:48 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:29:48 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:29:48 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:29:48 nebula gitea[2637956]:                 main: fn()
Oct 22 08:29:48 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:29:48 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:29:48 nebula gitea[2637956]:
Oct 22 08:29:48 nebula gitea[2637956]: 2020/10/22 08:29:48 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:29:49 nebula gitea[2637956]: 2020/10/22 08:29:49 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #2/15...
Oct 22 08:29:49 nebula gitea[2637956]: 2020/10/22 08:29:49 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:29:49 nebula gitea[2637956]: 2020/10/22 08:29:49 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:30:00 nebula gitea[2637956]: 2020/10/22 08:30:00 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #2/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:30:00 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:30:00 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:30:00 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:30:00 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:30:00 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:30:00 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:30:00 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:30:00 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:30:00 nebula gitea[2637956]:                 main: fn()
Oct 22 08:30:00 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:30:00 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:30:00 nebula gitea[2637956]:
Oct 22 08:30:00 nebula gitea[2637956]: 2020/10/22 08:30:00 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:30:01 nebula gitea[2637956]: 2020/10/22 08:30:01 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #3/15...
Oct 22 08:30:01 nebula gitea[2637956]: 2020/10/22 08:30:01 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:30:01 nebula gitea[2637956]: 2020/10/22 08:30:01 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:30:19 nebula gitea[2637956]: 2020/10/22 08:30:19 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #3/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:30:19 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:30:19 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:30:19 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:30:19 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:30:19 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:30:19 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:30:19 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:30:19 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:30:19 nebula gitea[2637956]:                 main: fn()
Oct 22 08:30:19 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:30:19 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:30:19 nebula gitea[2637956]:
Oct 22 08:30:19 nebula gitea[2637956]: 2020/10/22 08:30:19 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:30:20 nebula gitea[2637956]: 2020/10/22 08:30:20 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #4/15...
Oct 22 08:30:20 nebula gitea[2637956]: 2020/10/22 08:30:20 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:30:20 nebula gitea[2637956]: 2020/10/22 08:30:20 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:30:32 nebula gitea[2637956]: 2020/10/22 08:30:32 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #4/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:30:32 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:30:32 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:30:32 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:30:32 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:30:32 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:30:32 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:30:32 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:30:32 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:30:32 nebula gitea[2637956]:                 main: fn()
Oct 22 08:30:32 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:30:32 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:30:32 nebula gitea[2637956]:
Oct 22 08:30:32 nebula gitea[2637956]: 2020/10/22 08:30:32 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:30:33 nebula gitea[2637956]: 2020/10/22 08:30:33 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #5/15...
Oct 22 08:30:33 nebula gitea[2637956]: 2020/10/22 08:30:33 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:30:33 nebula gitea[2637956]: 2020/10/22 08:30:33 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:30:46 nebula gitea[2637956]: 2020/10/22 08:30:46 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #5/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:30:46 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:30:46 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:30:46 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:30:46 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:30:46 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:30:46 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:30:46 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:30:46 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:30:46 nebula gitea[2637956]:                 main: fn()
Oct 22 08:30:46 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:30:46 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:30:46 nebula gitea[2637956]:
Oct 22 08:30:46 nebula gitea[2637956]: 2020/10/22 08:30:46 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:30:47 nebula gitea[2637956]: 2020/10/22 08:30:47 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:30:47 nebula gitea[2637956]: 2020/10/22 08:30:47 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #6/15...
Oct 22 08:30:47 nebula gitea[2637956]: 2020/10/22 08:30:47 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:31:01 nebula gitea[2637956]: 2020/10/22 08:31:01 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #6/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:31:01 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:31:01 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:31:01 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:31:01 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:31:01 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:31:01 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:31:01 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:31:01 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:31:01 nebula gitea[2637956]:                 main: fn()
Oct 22 08:31:01 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:31:01 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:31:01 nebula gitea[2637956]:
Oct 22 08:31:01 nebula gitea[2637956]: 2020/10/22 08:31:01 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:31:02 nebula gitea[2637956]: 2020/10/22 08:31:02 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:31:02 nebula gitea[2637956]: 2020/10/22 08:31:02 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #7/15...
Oct 22 08:31:02 nebula gitea[2637956]: 2020/10/22 08:31:02 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:31:21 nebula gitea[2637956]: 2020/10/22 08:31:21 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #7/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:31:21 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:31:21 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:31:21 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:31:21 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:31:21 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:31:21 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:31:21 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:31:21 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:31:21 nebula gitea[2637956]:                 main: fn()
Oct 22 08:31:21 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:31:21 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:31:21 nebula gitea[2637956]:
Oct 22 08:31:21 nebula gitea[2637956]: 2020/10/22 08:31:21 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:31:22 nebula gitea[2637956]: 2020/10/22 08:31:22 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #8/15...
Oct 22 08:31:22 nebula gitea[2637956]: 2020/10/22 08:31:22 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:31:22 nebula gitea[2637956]: 2020/10/22 08:31:22 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:31:37 nebula gitea[2637956]: 2020/10/22 08:31:36 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #8/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:31:37 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:31:37 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:31:37 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:31:37 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:31:37 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:31:37 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:31:37 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:31:37 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:31:37 nebula gitea[2637956]:                 main: fn()
Oct 22 08:31:37 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:31:37 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:31:37 nebula gitea[2637956]:
Oct 22 08:31:37 nebula gitea[2637956]: 2020/10/22 08:31:36 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:31:38 nebula gitea[2637956]: 2020/10/22 08:31:38 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:31:38 nebula gitea[2637956]: 2020/10/22 08:31:38 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #9/15...
Oct 22 08:31:38 nebula gitea[2637956]: 2020/10/22 08:31:38 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:31:50 nebula gitea[2637956]: 2020/10/22 08:31:50 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #9/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:31:50 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:31:50 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:31:50 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:31:50 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:31:50 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:31:50 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:31:50 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:31:50 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:31:50 nebula gitea[2637956]:                 main: fn()
Oct 22 08:31:50 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:31:50 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:31:50 nebula gitea[2637956]:
Oct 22 08:31:50 nebula gitea[2637956]: 2020/10/22 08:31:50 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:31:51 nebula gitea[2637956]: 2020/10/22 08:31:51 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #10/15...
Oct 22 08:31:51 nebula gitea[2637956]: 2020/10/22 08:31:51 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:31:51 nebula gitea[2637956]: 2020/10/22 08:31:51 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:32:05 nebula gitea[2637956]: 2020/10/22 08:32:05 routers/init.go:85:initDBEngine() [E] ORM engine initialization attempt #10/15 failed. Error: migrate: do migrate: Repository 370 is not exist
Oct 22 08:32:05 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:85 (0x20395f3)
Oct 22 08:32:05 nebula gitea[2637956]:         /root/gitea-src/routers/init.go:177 (0x203a5b1)
Oct 22 08:32:05 nebula gitea[2637956]:         /root/gitea-src/cmd/web.go:153 (0x228669e)
Oct 22 08:32:05 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:526 (0x16c85bc)
Oct 22 08:32:05 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/command.go:174 (0x16c938d)
Oct 22 08:32:05 nebula gitea[2637956]:         /root/gitea-src/vendor/github.com/urfave/cli/app.go:279 (0x16c64a7)
Oct 22 08:32:05 nebula gitea[2637956]:         /root/gitea-src/main.go:110 (0x22c6ee9)
Oct 22 08:32:05 nebula gitea[2637956]:         /usr/lib/go/src/runtime/proc.go:204 (0x442aa8)
Oct 22 08:32:05 nebula gitea[2637956]:                 main: fn()
Oct 22 08:32:05 nebula gitea[2637956]:         /usr/lib/go/src/runtime/asm_amd64.s:1374 (0x478c20)
Oct 22 08:32:05 nebula gitea[2637956]:                 goexit: BYTE        $0x90        // NOP
Oct 22 08:32:05 nebula gitea[2637956]:
Oct 22 08:32:05 nebula gitea[2637956]: 2020/10/22 08:32:05 routers/init.go:86:initDBEngine() [I] Backing off for 1 seconds
Oct 22 08:32:06 nebula gitea[2637956]: 2020/10/22 08:32:06 routers/init.go:79:initDBEngine() [I] ORM engine initialization attempt #11/15...
Oct 22 08:32:06 nebula gitea[2637956]: 2020/10/22 08:32:06 ...rm/session_schema.go:25:Ping() [I] PING DATABASE sqlite3
Oct 22 08:32:06 nebula gitea[2637956]: 2020/10/22 08:32:06 ...ations/migrations.go:338:Migrate() [I] Migration[156]: fix publisher ID for tag releases
Oct 22 08:32:20 nebula systemd[1]: Stopping Gitea (Git with a cup of tea)...
```

## Description
I moved past the [commit with the subject migration](e9186362839749a83f4fcff2b01fc39de1ef4f45) (which has been approved in #6025 ) from a pretty recent [`commit`](327f18c3b28225dcfcaa0409b880136f7e1d1bf6).
Gitea failed to start properly as the migration failed with the above mentioned messages.
As the migration has not properly executed I could roll back.

### PR DESCRIPTION
**Pull request by lunny**
You don't know if the Gitea instance has how many releases which maybe over the capacity of the databases. So we should keep the transaction as a suitable size.

Also:

Fix #13255

### CODE REVIEW

**Comment by zeripath** [comment]
Also need to be nicer with missing repositories too

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
See: https://github.com/lunny/gitea/pull/9

File: models/migrations/v156.go [hunk 1]
```diff
@@ -50,15 +52,10 @@ func fixPublisherIDforTagReleases(x *xorm.Engine) error {
 	sess := x.NewSession()
 	defer sess.Close()
 
-	if err := sess.Begin(); err != nil {
-		return err
-	}
-
 	var (
 		gitRepoCache = make(map[int64]*git.Repository)
```
Comment (zeripath) [inline models/migrations/v156.go]: We may need to limit the size of this or... perhaps even better just order the releases by repository ID

**Comment by zeripath** [comment]
https://github.com/lunny/gitea/pull/10

**Review by zeripath — APPROVED** [review]

**Pull request merged**